apps/condo/domains/acquiring/routes/paymentLinkRouter.js

```diff
@@ -132,36 +158,35 @@ class PaymentLinkRouter {
             await this.checkLimits(req)
 
             if (isVirtual) {
-                const params = await this.extractVirtualReceiptParams(req)
+                const params = this.extractVirtualReceiptParams(req)
                 return await this.handlePaymentLink(
                     res,
                     params,
                     this.virtualReceiptPaymentValidator.bind(this),
-                    this.createMultiPaymentByVirtualReceipt.bind(this)
+                    this.createMultiPaymentByVirtualReceipt.bind(this),
                 )
-            } else {
+            } else if (this.isReceipt(req)) {
                 const params = this.extractRegularReceiptParams(req)
 
                 // for regular receipts we can check if billing receipt
                 // is already paid
-                const isAlreadyPaid = await this.checkReceiptAlreadyPaid(params)
-                if (isAlreadyPaid) {
-                    const { failureUrl } = params
-                    const errorPageUrl = `${conf.SERVER_URL}/500-error.html`
-                    const redirectUrl = new URL(failureUrl || errorPageUrl)
-
-                    // set common QP
-                    redirectUrl.searchParams.set('alreadyPaid', 'true')
-
-                    // redirect
-                    return res.redirect(redirectUrl.toString())
-                }
+                await this.redirectIfAlreadyPaid(res, params, this.checkReceiptAlreadyPaid.bind(this))
 
                 return await this.handlePaymentLink(
                     res,
                     params,
                     this.regularReceiptPaymentValidator.bind(this),
-                    this.createMultiPaymentByReceipt.bind(this)
+                    this.createMultiPaymentByReceipt.bind(this),
+                )
+            } else if (this.isInvoices(req)) {
+                const params = this.extractInvoicesParams(req)
+                await this.redirectIfAlreadyPaid(res, params, this.checkInvoicesAlreadyPaid.bind(this))
+
+                return await this.handlePaymentLink(
+                    res,
+                    params,
+                    this.invoicesPaymentValidator.bind(this),
+                    this.createMultiPaymentByInvoices.bind(this),
                 )
             }
```

Comment:
Probably we have to handle a case when no handlers chosen. Like redirect user to 500 page with some error message, like - we can not determinate a payment case

Reply:
Done